[package] qt5: error building for Android with OpenSSL

### Description

network module fails on the link step because it can't find ssl and crypto libs with special name (suffixed with `_arm64-v8a`), libs themselves do exist:

```
> ll /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/openssl/1.1.1s/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib
total 11624
drwxr-xr-x  3 kambala  staff    96B Jan 29 14:14 cmake/
drwxr-xr-x  2 kambala  staff    64B Jan 29 14:14 engines-1.1/
-rw-r--r--  1 kambala  staff   4.7M Jan 29 14:14 libcrypto.a
-rw-r--r--  1 kambala  staff   968K Jan 29 14:14 libssl.a
```

### Package and Environment Details

* Package Name/Version: **qt/5.15.8**
* Operating System+version: **macOS 12.6.2**
* Compiler+version: **clang 14**
* Conan version: **conan 1.57.0**
* Python version: **Python 3.10.9**


### Conan profile

Configuration (profile_host):
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.libcxx=c++_shared
compiler.version=14
os=Android
os.api_level=21
[options]
with_ffmpeg=False
[build_requires]
*: android-ndk/r24
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.cmake.cmaketoolchain:system_processor=aarch64

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=14
os=Macos
os_build=Macos
[options]
[build_requires]
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja

### Steps to reproduce

```
conan install qt/5.15.8@ \
  -o 'qt/*:openssl=True' \
  -o 'qt/*:opengl=es2' \
  -o 'qt/*:config=-android-sdk <path to SDK>' \
  -o 'qt/*:with_mysql=False' \
  -o 'qt/*:with_md4c=False' \
  -o 'qt/*:with_odbc=False' \
  -o 'qt/*:with_openal=False' \
  -o 'qt/*:with_pq=False'
```

### Logs

<details><summary>Click to expand log</summary>

```
Configuration (profile_host):
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.libcxx=c++_shared
compiler.version=14
os=Android
os.api_level=21
[options]
with_ffmpeg=False
[build_requires]
*: android-ndk/r24
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.cmake.cmaketoolchain:system_processor=aarch64

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=14
os=Macos
os_build=Macos
[options]
[build_requires]
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja

WARN: sdl_image/2.0.5: requirement sdl/2.0.20 overridden by your conanfile to sdl/2.26.1 
WARN: sdl_image/2.0.5: requirement libpng/1.6.37 overridden by your conanfile to libpng/1.6.39 
WARN: sdl_mixer/2.0.4: requirement sdl/2.0.20 overridden by your conanfile to sdl/2.26.1 
qt/5.15.8: WARN: Package binary is corrupted, removing: edd922a71b70a838b91c488616eb3304b186f418
Version ranges solved
    Version range '^1.69' required by 'conanfile.py' resolved to 'boost/1.81.0' in local cache
    Version range '~1.2.12' required by 'conanfile.py' resolved to 'minizip/1.2.13' in local cache
    Version range '^2021.3' required by 'conanfile.py' resolved to 'onetbb/2021.7.0' in local cache
    Version range '~5.15.2' required by 'conanfile.py' resolved to 'qt/5.15.8' in local cache
    Version range '~2.26.1 || >=2.0.20 <=2.22.0' required by 'conanfile.py' resolved to 'sdl/2.26.1' in local cache
    Version range '~2.0.5' required by 'conanfile.py' resolved to 'sdl_image/2.0.5' in local cache
    Version range '~2.0.4' required by 'conanfile.py' resolved to 'sdl_mixer/2.0.4' in local cache
    Version range '~2.0.18' required by 'conanfile.py' resolved to 'sdl_ttf/2.0.18' in local cache
    Version range '~1.6.38' required by 'conanfile.py' resolved to 'libpng/1.6.39' in local cache
    Version range '~1.2.13' required by 'conanfile.py' resolved to 'zlib/1.2.13' in local cache
    Version range '~1.17' required by 'conanfile.py' resolved to 'libiconv/1.17' in local cache

conanfile.py: Installing package
Requirements
    boost/1.81.0 from 'conancenter' - Cache
    brotli/1.0.9 from 'conancenter' - Cache
    bzip2/1.0.8 from 'conancenter' - Cache
    double-conversion/3.2.1 from 'conancenter' - Cache
    freetype/2.12.1 from 'conancenter' - Cache
    libiconv/1.17 from 'conancenter' - Cache
    libpng/1.6.39 from 'conancenter' - Cache
    minizip/1.2.13 from 'conancenter' - Cache
    mpg123/1.29.3 from 'conancenter' - Cache
    ogg/1.3.5 from 'conancenter' - Cache
    onetbb/2021.7.0 from 'conancenter' - Cache
    opengl/system from 'conancenter' - Cache
    openssl/1.1.1s from 'conancenter' - Cache
    pcre2/10.40 from 'conancenter' - Cache
    qt/5.15.8 from 'conancenter' - Cache
    sdl/2.26.1 from 'conancenter' - Cache
    sdl_image/2.0.5 from 'conancenter' - Cache
    sdl_mixer/2.0.4 from 'conancenter' - Cache
    sdl_ttf/2.0.18 from 'conancenter' - Cache
    sqlite3/3.39.4 from 'conancenter' - Cache
    vorbis/1.3.7 from 'conancenter' - Cache
    zlib/1.2.13 from 'conancenter' - Cache
    zstd/1.5.2 from 'conancenter' - Cache
Packages
    boost/1.81.0:cb6872ee80a534ab952c8fe76d7070930492f368 - Cache
    brotli/1.0.9:a14dda6584102a615ab8043bed7594e2cb807109 - Cache
    bzip2/1.0.8:2070bdc52264ee5d766c364d856228708d6cadbb - Cache
    double-conversion/3.2.1:ce2114a2d7957a097829dd5dd804c350f1a487b3 - Cache
    freetype/2.12.1:8eee413062ad3b049890854ae6798c205666a4d7 - Cache
    libiconv/1.17:8cbfc05d3e76d0a8b10531035a234407976a6e13 - Cache
    libpng/1.6.39:3bcd3ef0f6d66512f3b3820040087300d6f64ec8 - Cache
    minizip/1.2.13:3ae32ae34ef293a2af02e3bb35d4aca73931a9ff - Cache
    mpg123/1.29.3:854b8aa8ced9c0246df05f6a1c15eab0b4a972d0 - Cache
    ogg/1.3.5:8cbfc05d3e76d0a8b10531035a234407976a6e13 - Cache
    onetbb/2021.7.0:fbe8431dfa18a65f51bea687d84a5aeea16efe64 - Cache
    opengl/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    openssl/1.1.1s:8cbfc05d3e76d0a8b10531035a234407976a6e13 - Cache
    pcre2/10.40:83a2e622b7d61939b8619d4f36027d54394430d8 - Cache
    qt/5.15.8:edd922a71b70a838b91c488616eb3304b186f418 - Build
    sdl/2.26.1:4d15adc11081eaaea379639c8dcc4279dd49f773 - Cache
    sdl_image/2.0.5:7514bf17b735ea9b3263b50e48d8cec68fac3b7c - Build
    sdl_mixer/2.0.4:0668c0c6a3af913f9e52b0f2d4f38a5d1ec490cd - Build
    sdl_ttf/2.0.18:138e280c0daf533719d6d5de581f9e2afe6cd04b - Build
    sqlite3/3.39.4:c4cfd8dab27246e75aa1d0c8c8b2bcf1baa466eb - Cache
    vorbis/1.3.7:b2b741787e7bec41dd4e17b36ce7f1b4409e3f77 - Cache
    zlib/1.2.13:8cbfc05d3e76d0a8b10531035a234407976a6e13 - Cache
    zstd/1.5.2:01056e0f3f747bb87255bd28917fdbbaaaec5dfc - Cache
Build requirements
    android-ndk/r24 from 'conancenter' - Cache
Build requirements packages
    android-ndk/r24:06d313f2ce6737a31e5d7ccd4d516d2ac742276a - Cache

Cross-build from 'Macos:x86_64' to 'Android:armv8'
Installing (downloading, building) binaries...
android-ndk/r24: Already installed!
brotli/1.0.9: Already installed!
bzip2/1.0.8: Already installed!
double-conversion/3.2.1: Already installed!
libiconv/1.17: Already installed!
mpg123/1.29.3: Already installed!
mpg123/1.29.3: Appending PATH environment variable: /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/mpg123/1.29.3/_/_/package/854b8aa8ced9c0246df05f6a1c15eab0b4a972d0/bin
ogg/1.3.5: Already installed!
onetbb/2021.7.0: Already installed!
opengl/system: Already installed!
openssl/1.1.1s: Already installed!
pcre2/10.40: Already installed!
sqlite3/3.39.4: Already installed!
sqlite3/3.39.4: Appending PATH env var with : /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/sqlite3/3.39.4/_/_/package/c4cfd8dab27246e75aa1d0c8c8b2bcf1baa466eb/bin
zlib/1.2.13: Already installed!
zstd/1.5.2: Already installed!
boost/1.81.0: Already installed!
libpng/1.6.39: Already installed!
minizip/1.2.13: Already installed!
sdl/2.26.1: Already installed!
vorbis/1.3.7: Already installed!
freetype/2.12.1: Already installed!
qt/5.15.8: Applying build-requirement: android-ndk/r24
qt/5.15.8: WARN: Build folder is dirty, removing it: /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/build/edd922a71b70a838b91c488616eb3304b186f418
qt/5.15.8: Building your package in /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/build/edd922a71b70a838b91c488616eb3304b186f418
qt/5.15.8: Generator pkg_config created libcrypto.pc
qt/5.15.8: Generator pkg_config created libssl.pc
qt/5.15.8: Generator pkg_config created openssl.pc
qt/5.15.8: Generator pkg_config created libpcre2-32.pc
qt/5.15.8: Generator pkg_config created libpcre2-16.pc
qt/5.15.8: Generator pkg_config created libpcre2-8.pc
qt/5.15.8: Generator pkg_config created libpcre2-posix.pc
qt/5.15.8: Generator pkg_config created libpcre2.pc
qt/5.15.8: Generator pkg_config created double-conversion.pc
qt/5.15.8: Generator pkg_config created sqlite3.pc
qt/5.15.8: Generator pkg_config created opengl.pc
qt/5.15.8: Generator pkg_config created libzstd.pc
qt/5.15.8: Generator pkg_config created libpng.pc
qt/5.15.8: Generator pkg_config created zlib.pc
qt/5.15.8: Aggregating env generators
qt/5.15.8: Calling build()
+ cd qtbase
+ /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/source/qt5/qtbase/configure -top-level -shared -confirm-license -silent -nomake examples -nomake tests -prefix /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418 -extprefix /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418 -v -archdatadir /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418/bin/archdatadir -datadir /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418/bin/datadir -sysconfdir /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418/bin/sysconfdir -opensource -release -skip qtsvg -skip qtdeclarative -skip qtscript -skip qtmultimedia -skip qtxmlpatterns -skip qtdoc -skip qtlocation -skip qtsensors -skip qtconnectivity -skip qtwayland -skip qt3d -skip qtimageformats -skip qtgraphicaleffects -skip qtquickcontrols -skip qtserialbus -skip qtserialport -skip qtandroidextras -skip qtwebsockets -skip qtwebchannel -skip qtwebengine -skip qtwebview -skip qtquickcontrols2 -skip qtpurchasing -skip qtcharts -skip qtdatavis3d -skip qtvirtualkeyboard -skip qtgamepad -skip qtscxml -skip qtspeech -skip qtnetworkauth -skip qtremoteobjects -skip qtwebglplugin -skip qtlottie -skip qtquicktimeline -skip qtquick3d --zlib=system -opengl es2 -no-vulkan -openssl-linked --glib=no --pcre=system --fontconfig=no --icu=no --sql-mysql=no --sql-psql=no --sql-odbc=no --zstd=yes -no-dbus -no-feature-gssapi -system-doubleconversion -no-freetype -no-harfbuzz -no-libjpeg -system-libpng -system-sqlite -no-libmd4c ZLIB_LIBS=-lz OPENSSL_LIBS=-lssl -lcrypto PCRE2_LIBS=-lpcre2-posix -lpcre2-8 -lpcre2-16 -lpcre2-32 DOUBLECONVERSION_LIBS=-ldouble-conversion LIBPNG_LIBS=-lpng -lm -lz SQLITE_LIBS=-lsqlite3 ZSTD_LIBS=-lzstd -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/openssl/1.1.1s/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/pcre2/10.40/_/_/package/83a2e622b7d61939b8619d4f36027d54394430d8/include -D PCRE2_STATIC -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/double-conversion/3.2.1/_/_/package/ce2114a2d7957a097829dd5dd804c350f1a487b3/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/sqlite3/3.39.4/_/_/package/c4cfd8dab27246e75aa1d0c8c8b2bcf1baa466eb/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zstd/1.5.2/_/_/package/01056e0f3f747bb87255bd28917fdbbaaaec5dfc/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/libpng/1.6.39/_/_/package/3bcd3ef0f6d66512f3b3820040087300d6f64ec8/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zlib/1.2.13/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/include QMAKE_LIBDIR+=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/openssl/1.1.1s/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/pcre2/10.40/_/_/package/83a2e622b7d61939b8619d4f36027d54394430d8/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/double-conversion/3.2.1/_/_/package/ce2114a2d7957a097829dd5dd804c350f1a487b3/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/sqlite3/3.39.4/_/_/package/c4cfd8dab27246e75aa1d0c8c8b2bcf1baa466eb/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zstd/1.5.2/_/_/package/01056e0f3f747bb87255bd28917fdbbaaaec5dfc/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/libpng/1.6.39/_/_/package/3bcd3ef0f6d66512f3b3820040087300d6f64ec8/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zlib/1.2.13/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib QMAKE_RPATHLINKDIR+=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/openssl/1.1.1s/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/pcre2/10.40/_/_/package/83a2e622b7d61939b8619d4f36027d54394430d8/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/double-conversion/3.2.1/_/_/package/ce2114a2d7957a097829dd5dd804c350f1a487b3/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/sqlite3/3.39.4/_/_/package/c4cfd8dab27246e75aa1d0c8c8b2bcf1baa466eb/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zstd/1.5.2/_/_/package/01056e0f3f747bb87255bd28917fdbbaaaec5dfc/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/libpng/1.6.39/_/_/package/3bcd3ef0f6d66512f3b3820040087300d6f64ec8/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zlib/1.2.13/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib -android-ndk-platform android-21 -android-abis arm64-v8a -xplatform android-clang QMAKE_CC=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang QMAKE_LINK_C=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang QMAKE_LINK_C_SHLIB=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang QMAKE_CXX=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang++ QMAKE_LINK=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang++ QMAKE_LINK_SHLIB=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang++ -no-gif -no-ico -no-feature-imageformat_bmp -no-feature-imageformat_jpeg -no-feature-imageformat_ppm -no-feature-imageformat_xbm -android-sdk /Users/kambala/dev/android/sdk
Performing shadow build...
Preparing build tree...
Creating qmake...
Info: creating cache file /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/build/edd922a71b70a838b91c488616eb3304b186f418/build_folder/.qmake.cache
Command line: -shared -confirm-license -silent -nomake examples -nomake tests -prefix /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418 -extprefix /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418 -v -archdatadir /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418/bin/archdatadir -datadir /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418/bin/datadir -sysconfdir /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/qt/5.15.8/_/_/package/edd922a71b70a838b91c488616eb3304b186f418/bin/sysconfdir -opensource -release -skip qtsvg -skip qtdeclarative -skip qtscript -skip qtmultimedia -skip qtxmlpatterns -skip qtdoc -skip qtlocation -skip qtsensors -skip qtconnectivity -skip qtwayland -skip qt3d -skip qtimageformats -skip qtgraphicaleffects -skip qtquickcontrols -skip qtserialbus -skip qtserialport -skip qtandroidextras -skip qtwebsockets -skip qtwebchannel -skip qtwebengine -skip qtwebview -skip qtquickcontrols2 -skip qtpurchasing -skip qtcharts -skip qtdatavis3d -skip qtvirtualkeyboard -skip qtgamepad -skip qtscxml -skip qtspeech -skip qtnetworkauth -skip qtremoteobjects -skip qtwebglplugin -skip qtlottie -skip qtquicktimeline -skip qtquick3d --zlib=system -opengl es2 -no-vulkan -openssl-linked --glib=no --pcre=system --fontconfig=no --icu=no --sql-mysql=no --sql-psql=no --sql-odbc=no --zstd=yes -no-dbus -no-feature-gssapi -system-doubleconversion -no-freetype -no-harfbuzz -no-libjpeg -system-libpng -system-sqlite -no-libmd4c ZLIB_LIBS=-lz 'OPENSSL_LIBS=-lssl -lcrypto' 'PCRE2_LIBS=-lpcre2-posix -lpcre2-8 -lpcre2-16 -lpcre2-32' DOUBLECONVERSION_LIBS=-ldouble-conversion 'LIBPNG_LIBS=-lpng -lm -lz' SQLITE_LIBS=-lsqlite3 ZSTD_LIBS=-lzstd -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/openssl/1.1.1s/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/pcre2/10.40/_/_/package/83a2e622b7d61939b8619d4f36027d54394430d8/include -D PCRE2_STATIC -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/double-conversion/3.2.1/_/_/package/ce2114a2d7957a097829dd5dd804c350f1a487b3/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/sqlite3/3.39.4/_/_/package/c4cfd8dab27246e75aa1d0c8c8b2bcf1baa466eb/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zstd/1.5.2/_/_/package/01056e0f3f747bb87255bd28917fdbbaaaec5dfc/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/libpng/1.6.39/_/_/package/3bcd3ef0f6d66512f3b3820040087300d6f64ec8/include -I /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zlib/1.2.13/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/include 'QMAKE_LIBDIR+=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/openssl/1.1.1s/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/pcre2/10.40/_/_/package/83a2e622b7d61939b8619d4f36027d54394430d8/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/double-conversion/3.2.1/_/_/package/ce2114a2d7957a097829dd5dd804c350f1a487b3/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/sqlite3/3.39.4/_/_/package/c4cfd8dab27246e75aa1d0c8c8b2bcf1baa466eb/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zstd/1.5.2/_/_/package/01056e0f3f747bb87255bd28917fdbbaaaec5dfc/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/libpng/1.6.39/_/_/package/3bcd3ef0f6d66512f3b3820040087300d6f64ec8/lib /Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zlib/1.2.13/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib' QMAKE_RPATHLINKDIR+=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/openssl/1.1.1s/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/pcre2/10.40/_/_/package/83a2e622b7d61939b8619d4f36027d54394430d8/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/double-conversion/3.2.1/_/_/package/ce2114a2d7957a097829dd5dd804c350f1a487b3/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/sqlite3/3.39.4/_/_/package/c4cfd8dab27246e75aa1d0c8c8b2bcf1baa466eb/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zstd/1.5.2/_/_/package/01056e0f3f747bb87255bd28917fdbbaaaec5dfc/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/libpng/1.6.39/_/_/package/3bcd3ef0f6d66512f3b3820040087300d6f64ec8/lib:/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/zlib/1.2.13/_/_/package/8cbfc05d3e76d0a8b10531035a234407976a6e13/lib -android-ndk-platform android-21 -android-abis arm64-v8a -xplatform android-clang QMAKE_CC=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang QMAKE_LINK_C=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang QMAKE_LINK_C_SHLIB=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang QMAKE_CXX=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang++ QMAKE_LINK=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang++ QMAKE_LINK_SHLIB=/Users/kambala/dev/vcmi/conan-deps-android-64/.conan/data/android-ndk/r24/_/_/package/06d313f2ce6737a31e5d7ccd4d516d2ac742276a/bin/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android21-clang++ -no-gif -no-ico -no-feature-imageformat_bmp -no-feature-imageformat_jpeg -no-feature-imageformat_ppm -no-feature-imageformat_xbm -android-sdk /Users/kambala/dev/android/sdk

This is the Qt Open Source Edition.

You have already accepted the terms of the Open Source license.

Running configuration tests...

...

Configure summary:

Building on: macx-clang (x86_64, CPU features: cx16 mmx sse sse2 sse3 ssse3 sse4.1)
Building for: android-clang (arm64, CPU features: neon)
Target compiler: clang (Apple) 14.0.1
Configuration: cross_compile compile_examples enable_new_dtags neon precompile_header silent shared shared release c++11 c++14 c++17 c++1z concurrent no-pkg-config reduce_exports stl
Build options:
  Mode ................................... release
  Optimize release build for size ........ no
  Building shared libraries .............. yes
  Using C standard ....................... C11
  Using C++ standard ..................... C++17
  Using ccache ........................... no
  Using new DTAGS ........................ yes
  Relocatable ............................ yes
  Using precompiled headers .............. yes
  Using LTCG ............................. no
  Target compiler supports:
    NEON ................................. yes
  Build parts ............................ libs
  App store compliance ................... yes
Qt modules and options:
  Qt Concurrent .......................... yes
  Qt D-Bus ............................... no
  Qt D-Bus directly linked to libdbus .... no
  Qt Gui ................................. yes
  Qt Network ............................. yes
  Qt Sql ................................. yes
  Qt Testlib ............................. yes
  Qt Widgets ............................. yes
  Qt Xml ................................. yes
Support enabled for:
  Using pkg-config ....................... no
  udev ................................... no
  Using system zlib ...................... yes
  Zstandard support ...................... yes
Qt Core:
  DoubleConversion ....................... yes
    Using system DoubleConversion ........ yes
  GLib ................................... no
  iconv .................................. no
  ICU .................................... no
  Built-in copy of the MIME database ..... yes
  Tracing backend ........................ <none>
  Logging backends:
    journald ............................. no
    syslog ............................... no
    slog2 ................................ no
  PCRE2 .................................. yes
    Using system PCRE2 ................... yes
Qt Network:
  getifaddrs() ........................... no
  IPv6 ifname ............................ no
  libproxy ............................... no
  Linux AF_NETLINK ....................... no
  OpenSSL ................................ yes
    Qt directly linked to OpenSSL ........ yes
  OpenSSL 1.1 ............................ yes
  DTLS ................................... yes
  OCSP-stapling .......................... yes
  SCTP ................................... no
  Use system proxies ..................... yes
  GSSAPI ................................. no
Qt Gui:
  Accessibility .......................... yes
  FreeType ............................... no
    Using system FreeType ................ no
  HarfBuzz ............................... no
    Using system HarfBuzz ................ no
  Fontconfig ............................. no
  Image formats:
    GIF .................................. no
    ICO .................................. no
    JPEG ................................. no
      Using system libjpeg ............... no
    PNG .................................. yes
      Using system libpng ................ yes
  Text formats:
    HtmlParser ........................... yes
    CssParser ............................ yes
    OdfWriter ............................ yes
    MarkdownReader ....................... no
      Using system libmd4c ............... no
    MarkdownWriter ....................... yes
  EGL .................................... yes
  OpenVG ................................. no
  OpenGL:
    Desktop OpenGL ....................... no
    OpenGL ES 2.0 ........................ yes
    OpenGL ES 3.0 ........................ no
    OpenGL ES 3.1 ........................ no
    OpenGL ES 3.2 ........................ no
  Vulkan ................................. no
  Session Management ..................... yes
Features used by QPA backends:
  evdev .................................. yes
  libinput ............................... no
  INTEGRITY HID .......................... no
  mtdev .................................. no
  tslib .................................. no
  xkbcommon .............................. no
  X11 specific:
    XLib ................................. no
    XCB Xlib ............................. no
    EGL on X11 ........................... no
    xkbcommon-x11 ........................ no
QPA backends:
  DirectFB ............................... no
  EGLFS .................................. no
  LinuxFB ................................ no
  VNC .................................... no
Qt Sql:
  SQL item models ........................ yes
Qt Widgets:
  GTK+ ................................... no
  Styles ................................. Fusion Windows Android
Qt PrintSupport:
  CUPS ................................... no
Qt Sql Drivers:
  DB2 (IBM) .............................. no
  InterBase .............................. no
  MySql .................................. no
  OCI (Oracle) ........................... no
  ODBC ................................... no
  PostgreSQL ............................. no
  SQLite2 ................................ no
  SQLite ................................. yes
    Using system provided SQLite ......... yes
  TDS (Sybase) ........................... no
Qt Testlib:
  Tester for item models ................. yes
Qt Tools:
  Qt Assistant ........................... yes
  Qt Designer ............................ yes
  Qt Distance Field Generator ............ yes
  kmap2qmap .............................. yes
  Qt Linguist ............................ yes
  Mac Deployment Tool .................... no
  makeqpf ................................ yes
  pixeltool .............................. yes
  qdbus .................................. yes
  qev .................................... yes
  Qt Attributions Scanner ................ yes
  qtdiag ................................. yes
  qtpaths ................................ yes
  qtplugininfo ........................... yes
  Windows deployment tool ................ no
  WinRT Runner Tool ...................... no
Qt Tools:
  QDoc ................................... no

WARNING: Cross compiling without sysroot. Disabling pkg-config

WARNING: QDoc will not be compiled, probably because libclang could not be located. This means that you cannot build the Qt documentation.

...

linking ../../lib/libQt5Network_arm64-v8a.so
ld: error: unable to find library -lssl_arm64-v8a
ld: error: unable to find library -lcrypto_arm64-v8a
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [../../lib/libQt5Network_arm64-v8a.so] Error 1
make[2]: *** [sub-network-make_first] Error 2
make[2]: *** Waiting for unfinished jobs....
```

</details>

--- a/recipes/qt/5.x.x/conandata.yml
+++ b/recipes/qt/5.x.x/conandata.yml
@@ -45,6 +45,8 @@ patches:
       base_path: "qt5/qtwebengine/src/3rdparty"
     - patch_file: "patches/337f28c9ab-5.15.8.patch"
       base_path: "qt5/qtbase"
+    - patch_file: "patches/android-backtrace.diff"
+      base_path: "qt5/qtbase"
   "5.15.7":
     - patch_file: "patches/337f28c9ab.patch"
       base_path: "qt5/qtbase"
@@ -70,6 +72,8 @@ patches:
       base_path: "qt5/qtwebengine/src/3rdparty/chromium/third_party/skia"
     - patch_file: "patches/0001-Find-fontconfig-using-pkg-config.patch"
       base_path: "qt5/qtwebengine/src/3rdparty"
+    - patch_file: "patches/android-backtrace.diff"
+      base_path: "qt5/qtbase"
   "5.15.6":
     - patch_file: "patches/337f28c9ab.patch"
       base_path: "qt5/qtbase"
@@ -95,6 +99,8 @@ patches:
       base_path: "qt5/qtwebengine/src/3rdparty/chromium/third_party/skia"
     - patch_file: "patches/0001-Find-fontconfig-using-pkg-config.patch"
       base_path: "qt5/qtwebengine/src/3rdparty"
+    - patch_file: "patches/android-backtrace.diff"
+      base_path: "qt5/qtbase"
   "5.15.5":
     - patch_file: "patches/337f28c9ab-5.15.5.patch"
       base_path: "qt5/qtbase"
@@ -120,3 +126,5 @@ patches:
       base_path: "qt5/qtwebengine/src/3rdparty/chromium/third_party/skia"
     - patch_file: "patches/0001-Find-fontconfig-using-pkg-config.patch"
       base_path: "qt5/qtwebengine/src/3rdparty"
+    - patch_file: "patches/android-backtrace.diff"
+      base_path: "qt5/qtbase"

--- a/recipes/qt/5.x.x/conandata.yml
+++ b/recipes/qt/5.x.x/conandata.yml
@@ -47,6 +47,8 @@ patches:
       base_path: "qt5/qtbase"
     - patch_file: "patches/android-backtrace.diff"
       base_path: "qt5/qtbase"
+    - patch_file: "patches/android-openssl.diff"
+      base_path: "qt5/qtbase"
   "5.15.7":
     - patch_file: "patches/337f28c9ab.patch"
       base_path: "qt5/qtbase"
@@ -73,6 +75,8 @@ patches:
     - patch_file: "patches/0001-Find-fontconfig-using-pkg-config.patch"
       base_path: "qt5/qtwebengine/src/3rdparty"
     - patch_file: "patches/android-backtrace.diff"
+      base_path: "qt5/qtbase"
+    - patch_file: "patches/android-openssl.diff"
       base_path: "qt5/qtbase"
   "5.15.6":
     - patch_file: "patches/337f28c9ab.patch"
@@ -101,6 +105,8 @@ patches:
       base_path: "qt5/qtwebengine/src/3rdparty"
     - patch_file: "patches/android-backtrace.diff"
       base_path: "qt5/qtbase"
+    - patch_file: "patches/android-openssl.diff"
+      base_path: "qt5/qtbase"
   "5.15.5":
     - patch_file: "patches/337f28c9ab-5.15.5.patch"
       base_path: "qt5/qtbase"
@@ -127,4 +133,6 @@ patches:
     - patch_file: "patches/0001-Find-fontconfig-using-pkg-config.patch"
       base_path: "qt5/qtwebengine/src/3rdparty"
     - patch_file: "patches/android-backtrace.diff"
+      base_path: "qt5/qtbase"
+    - patch_file: "patches/android-openssl.diff"
       base_path: "qt5/qtbase"

--- a/recipes/qt/5.x.x/conandata.yml
+++ b/recipes/qt/5.x.x/conandata.yml
@@ -49,6 +49,8 @@ patches:
       base_path: "qt5/qtbase"
     - patch_file: "patches/android-openssl.diff"
       base_path: "qt5/qtbase"
+    - patch_file: "patches/android-new-ndk.diff"
+      base_path: "qt5/qtbase"
   "5.15.7":
     - patch_file: "patches/337f28c9ab.patch"
       base_path: "qt5/qtbase"
@@ -77,6 +79,8 @@ patches:
     - patch_file: "patches/android-backtrace.diff"
       base_path: "qt5/qtbase"
     - patch_file: "patches/android-openssl.diff"
+      base_path: "qt5/qtbase"
+    - patch_file: "patches/android-new-ndk.diff"
       base_path: "qt5/qtbase"
   "5.15.6":
     - patch_file: "patches/337f28c9ab.patch"
@@ -107,6 +111,8 @@ patches:
       base_path: "qt5/qtbase"
     - patch_file: "patches/android-openssl.diff"
       base_path: "qt5/qtbase"
+    - patch_file: "patches/android-new-ndk.diff"
+      base_path: "qt5/qtbase"
   "5.15.5":
     - patch_file: "patches/337f28c9ab-5.15.5.patch"
       base_path: "qt5/qtbase"
@@ -135,4 +141,6 @@ patches:
     - patch_file: "patches/android-backtrace.diff"
       base_path: "qt5/qtbase"
     - patch_file: "patches/android-openssl.diff"
+      base_path: "qt5/qtbase"
+    - patch_file: "patches/android-new-ndk.diff"
       base_path: "qt5/qtbase"

--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -312,7 +312,7 @@ class QtConan(ConanFile):
         if self.options.get_safe("qtwayland", False) and not self.dependencies.direct_host["xkbcommon"].options.with_wayland:
             raise ConanInvalidConfiguration("The 'with_wayland' option for the 'xkbcommon' package must be enabled when the 'qtwayland' option is enabled")
 
-        if cross_building(self) and self.options.cross_compile == "None" and not is_apple_os(self):
+        if cross_building(self) and self.options.cross_compile == "None" and not is_apple_os(self) and self.settings.os != "Android":
             raise ConanInvalidConfiguration("option cross_compile must be set for cross compilation "
                                             "cf https://doc.qt.io/qt-5/configure-options.html#cross-compilation-options")
 

--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -1429,8 +1429,9 @@ Examples = bin/datadir/examples""")
                     self.cpp_info.components["qtNetwork"].frameworks.append("GSS")
                 if not self.options.openssl: # with SecureTransport
                     self.cpp_info.components["qtNetwork"].frameworks.append("Security")
-            if self.settings.os == "Macos":
+            if self.settings.os == "Macos" or (self.settings.os == "iOS" and Version(self.settings.compiler.version) >= "14.0"):
                 self.cpp_info.components["qtCore"].frameworks.append("IOKit")     # qtcore requires "_IORegistryEntryCreateCFProperty", "_IOServiceGetMatchingService" and much more which are in "IOKit" framework
+            if self.settings.os == "Macos":
                 self.cpp_info.components["qtCore"].frameworks.append("Cocoa")     # qtcore requires "_OBJC_CLASS_$_NSApplication" and more, which are in "Cocoa" framework
                 self.cpp_info.components["qtCore"].frameworks.append("Security")  # qtcore requires "_SecRequirementCreateWithString" and more, which are in "Security" framework
 

--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -1,5 +1,6 @@
 from conan import ConanFile
 from conan.errors import ConanInvalidConfiguration
+from conan.tools.android import android_abi
 from conan.tools.apple import is_apple_os
 from conan.tools.build import build_jobs, check_min_cppstd, cross_building
 from conan.tools.env import Environment, VirtualBuildEnv, VirtualRunEnv
@@ -14,7 +15,7 @@ import os
 import textwrap
 import shutil
 
-required_conan_version = ">=1.52.0"
+required_conan_version = ">=1.59.0"
 
 
 class QtConan(ConanFile):
@@ -969,6 +970,8 @@ Examples = bin/datadir/examples""")
             self.cpp_info.components[component].build_modules["cmake_find_package_multi"].append(module)
 
         libsuffix = ""
+        if self.settings.os == "Android":
+            libsuffix = f"_{android_abi(self)}"
         if not self.options.multiconfiguration:
             if self.settings.build_type == "Debug":
                 if self.settings.os == "Windows" and is_msvc(self):

--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -69,6 +69,7 @@ class QtConan(ConanFile):
         "gui": [True, False],
         "widgets": [True, False],
 
+        "android_sdk": [None, "ANY"],
         "device": [None, "ANY"],
         "cross_compile": [None, "ANY"],
         "sysroot": [None, "ANY"],
@@ -111,6 +112,7 @@ class QtConan(ConanFile):
         "gui": True,
         "widgets": True,
 
+        "android_sdk": None,
         "device": None,
         "cross_compile": None,
         "sysroot": None,
@@ -187,6 +189,9 @@ class QtConan(ConanFile):
 
         if self.settings.os != "Macos":
             del self.options.qtmacextras
+
+        if self.settings.os != "Android":
+            del self.options.android_sdk
 
     def configure(self):
         # if self.settings.os != "Linux":
@@ -276,8 +281,11 @@ class QtConan(ConanFile):
             if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "5":
                 raise ConanInvalidConfiguration("Compiling Qt WebEngine with gcc < 5 is not supported")
 
-        if self.settings.os == "Android" and self.options.get_safe("opengl", "no") == "desktop":
-            raise ConanInvalidConfiguration("OpenGL desktop is not supported on Android. Consider using OpenGL es2")
+        if self.settings.os == "Android":
+            if self.options.get_safe("opengl", "no") == "desktop":
+                raise ConanInvalidConfiguration("OpenGL desktop is not supported on Android. Consider using OpenGL es2")
+            if not self.options.get_safe("android_sdk", ""):
+                raise ConanInvalidConfiguration("Path to Android SDK is required to build Qt")
 
         if self.settings.os != "Windows" and self.options.get_safe("opengl", "no") == "dynamic":
             raise ConanInvalidConfiguration("Dynamic OpenGL is supported only on Windows.")
@@ -736,6 +744,8 @@ class QtConan(ConanFile):
         elif self.settings.get_safe("compiler.libcxx") == "libstdc++11":
             args += ["-D_GLIBCXX_USE_CXX11_ABI=1"]
 
+        if self.options.get_safe("android_sdk", ""):
+            args += [f"-android-sdk {self.options.android_sdk}"]
         if self.options.sysroot:
             args += [f"-sysroot {self.options.sysroot}"]
 
@@ -940,6 +950,8 @@ Examples = bin/datadir/examples""")
                     self.info.settings.compiler.runtime = "MT/MTd"
             else:
                 self.info.settings.compiler.runtime_type = "Release/Debug"
+        if self.settings.os == "Android":
+            del self.info.options.android_sdk
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "Qt5")

--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -735,10 +735,7 @@ class QtConan(ConanFile):
                 args.append('QMAKE_APPLE_DEVICE_ARCHS="arm64"')
         elif self.settings.os == "Android":
             args += [f"-android-ndk-platform android-{self.settings.os.api_level}"]
-            args += ["-android-abis %s" % {"armv7": "armeabi-v7a",
-                                           "armv8": "arm64-v8a",
-                                           "x86": "x86",
-                                           "x86_64": "x86_64"}.get(str(self.settings.arch))]
+            args += [f"-android-abis {android_abi(self)}"]
 
         if self.settings.get_safe("compiler.libcxx") == "libstdc++":
             args += ["-D_GLIBCXX_USE_CXX11_ABI=0"]

--- a/recipes/qt/5.x.x/patches/android-backtrace.diff
+++ b/recipes/qt/5.x.x/patches/android-backtrace.diff
@@ -1,0 +1,19 @@
+diff --git a/src/corelib/global/qlogging.cpp b/src/corelib/global/qlogging.cpp
+index 89f49324c..dedf78fe5 100644
+--- a/src/corelib/global/qlogging.cpp
++++ b/src/corelib/global/qlogging.cpp
+@@ -107,7 +107,13 @@
+ #      define QLOGGING_HAVE_BACKTRACE
+ #    endif
+ #  elif (defined(__GLIBC__) && defined(__GLIBCXX__)) || (__has_include(<cxxabi.h>) && __has_include(<execinfo.h>))
+-#    define QLOGGING_HAVE_BACKTRACE
++#    ifdef __ANDROID_API__
++#      if __ANDROID_API__ >= 33
++#        define QLOGGING_HAVE_BACKTRACE
++#      endif
++#    else
++#      define QLOGGING_HAVE_BACKTRACE
++#    endif
+ #  endif
+ #endif
+ 

--- a/recipes/qt/5.x.x/patches/android-new-ndk.diff
+++ b/recipes/qt/5.x.x/patches/android-new-ndk.diff
@@ -1,0 +1,27 @@
+From 49f62e8c5a8aef4c0b7fd6867a30e653d2f5dd98 Mon Sep 17 00:00:00 2001
+From: Alexey Edelev <alexey.edelev@qt.io>
+Date: Mon, 21 Nov 2022 15:03:17 +0100
+Subject: [PATCH] Remove QMAKE_RANLIB and QMAKE_LINK_SHLIB from android/default_pre.prf
+
+The values should come from android-clang mkspecs. The hardcoded
+values don't work correctly with recent Android NDKs.
+
+Pick-to: 6.4 6.2 5.15
+Fixes: QTBUG-108662
+Change-Id: Ie153a50ee0c49bd4f0704b588a4e2c87a05c1063
+Reviewed-by: Assam Boudjelthia <assam.boudjelthia@qt.io>
+---
+
+diff --git a/mkspecs/features/android/default_pre.prf b/mkspecs/features/android/default_pre.prf
+index 9f90dcb..2328b72 100644
+--- a/mkspecs/features/android/default_pre.prf
++++ b/mkspecs/features/android/default_pre.prf
+@@ -76,8 +76,6 @@
+ else: equals(QT_ARCH, arm64-v8a): CROSS_COMPILE = $$NDK_LLVM_PATH/bin/aarch64-linux-android-
+ else: CROSS_COMPILE = $$NDK_LLVM_PATH/bin/arm-linux-androideabi-
+ 
+-QMAKE_RANLIB            = $${CROSS_COMPILE}ranlib
+-QMAKE_LINK_SHLIB        = $$QMAKE_LINK
+ QMAKE_LFLAGS            =
+ 
+ QMAKE_LIBS_PRIVATE      = -llog -lz -lm -ldl -lc

--- a/recipes/qt/5.x.x/patches/android-openssl.diff
+++ b/recipes/qt/5.x.x/patches/android-openssl.diff
@@ -1,0 +1,13 @@
+diff --git a/src/network/ssl/ssl.pri b/src/network/ssl/ssl.pri
+index 230c45c26..d7527949e 100644
+--- a/src/network/ssl/ssl.pri
++++ b/src/network/ssl/ssl.pri
+@@ -117,7 +117,7 @@ qtConfig(ssl) {
+ 
+         qtConfig(openssl-linked): {
+             android {
+-                build_pass|single_android_abi: LIBS_PRIVATE += -lssl_$${QT_ARCH} -lcrypto_$${QT_ARCH}
++                build_pass|single_android_abi: LIBS_PRIVATE += -lssl -lcrypto
+             } else: QMAKE_USE_FOR_PRIVATE += openssl
+         } else: \
+             QMAKE_USE_FOR_PRIVATE += openssl/nolink

